File: app/src/main/java/marabillas/loremar/lmvideodownloader/RenameDialog.java
/*
 *     LM videodownloader is a browser app for android, made to easily
 *     download videos.
 *     Copyright (C) 2018 Loremar Marabillas
 *
 *     This program is free software; you can redistribute it and/or modify
 *     it under the terms of the GNU General Public License as published by
 *     the Free Software Foundation; either version 2 of the License, or
 *     (at your option) any later version.
 *
 *     This program is distributed in the hope that it will be useful,
 *     but WITHOUT ANY WARRANTY; without even the implied warranty of
 *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *     GNU General Public License for more details.
 *
 *     You should have received a copy of the GNU General Public License along
 *     with this program; if not, write to the Free Software Foundation, Inc.,
 *     51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 */

package marabillas.loremar.lmvideodownloader;

import android.app.AlertDialog;
import android.content.Context;
import android.content.DialogInterface;
import android.view.ViewGroup;
import android.widget.EditText;

/**
 * Created by loremar on 3/20/18.
 *
 */

public abstract class RenameDialog implements DialogInterface.OnClickListener {
    private EditText text;

    RenameDialog(Context context, String hint) {
        AlertDialog dialog = new AlertDialog.Builder(context).create();
        text = new EditText(context);
        text.setLayoutParams(new ViewGroup.LayoutParams(ViewGroup.LayoutParams.WRAP_CONTENT,
                ViewGroup.LayoutParams.WRAP_CONTENT));
        text.setHint(hint);
        dialog.setView(text);
        dialog.setMessage("Type new name:");
        dialog.setButton(DialogInterface.BUTTON_POSITIVE, "OK", this);
        dialog.setButton(DialogInterface.BUTTON_NEGATIVE, "CANCEL", this);
        dialog.show();
    }

    @Override
    public final void onClick(DialogInterface dialog, int which) {
        if(which == DialogInterface.BUTTON_POSITIVE) {
            onOK(text.getText().toString());
        }
    }

    abstract void onOK(String newName);
}

File: features/browser/src/main/java/marabillas/loremar/beedio/browser/views/RenameDialog.kt
/*
 * Beedio is an Android app for downloading videos
 * Copyright (C) 2019 Loremar Marabillas
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License along
 * with this program; if not, write to the Free Software Foundation, Inc.,
 * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 */

package marabillas.loremar.beedio.browser.views

import android.content.Context
import android.view.ViewGroup
import android.view.ViewGroup.LayoutParams.MATCH_PARENT
import android.view.ViewGroup.LayoutParams.WRAP_CONTENT
import android.view.inputmethod.EditorInfo
import android.widget.EditText
import androidx.appcompat.app.AlertDialog
import marabillas.loremar.beedio.browser.R

class RenameDialog(context: Context, private val doOnDone: (String) -> Unit) {
    private lateinit var dialog: AlertDialog
    private val editText: EditText

    init {
        editText = EditText(context).apply {
            layoutParams = ViewGroup.LayoutParams(MATCH_PARENT, WRAP_CONTENT)
            hint = context.getString(R.string.enter_new_title)
            inputType = EditorInfo.TYPE_CLASS_TEXT
            setOnEditorActionListener { _, actionId, _ ->
                if (actionId == EditorInfo.IME_ACTION_DONE && processInput()) {
                    dialog.dismiss()
                    true
                } else {
                    false
                }
            }
        }

        dialog = AlertDialog.Builder(context)
                .setTitle("Rename")
                .setPositiveButton("DONE", null)
                .setNegativeButton("CANCEL", null)
                .setView(editText)
                .create()


        dialog.apply {
            setOnShowListener {
                dialog.getButton(AlertDialog.BUTTON_POSITIVE).setOnClickListener {
                    if (processInput()) dialog.dismiss()
                }
            }
            show()
        }
    }

    private fun processInput(): Boolean {
        val input = editText.text.toString()
        return if (input.isBlank()) {
            editText.error = "Text input can not be blank."
            false
        } else {
            doOnDone(input)
            true
        }
    }
}